x/crosschain/keeper/msg_server_add_to_outtx_tracker.go

```diff
@@ -36,180 +34,113 @@ func (k msgServer) AddToOutTxTracker(goCtx context.Context, msg *types.MsgAddToO
+		return types.ErrProofVerificationFail.Wrapf(err.Error())
+	}
+
 	// get tss address
 	var bitcoinChainID int64
 	if chains.IsBitcoinChain(msg.ChainId) {
 		bitcoinChainID = msg.ChainId
 	}
-	tss, err := k.zetaObserverKeeper.GetTssAddress(ctx, &observertypes.QueryGetTssAddressRequest{
+
+	tss, err := k.GetObserverKeeper().GetTssAddress(ctx, &observertypes.QueryGetTssAddressRequest{
 		BitcoinChainId: bitcoinChainID,
 	})
-	if err != nil {
-		return err
-	}
-
-	// verify message against transaction body
-	if chains.IsEVMChain(msg.ChainId) {
-		err = VerifyEVMOutTxBody(msg, txBytes, tss.Eth)
-	} else if chains.IsBitcoinChain(msg.ChainId) {
-		err = VerifyBTCOutTxBody(msg, txBytes, tss.Btc)
-	} else {
-		return fmt.Errorf("cannot verify outTx body for chain %d", msg.ChainId)
-	}
-	return err
-}
-
-// VerifyEVMOutTxBody validates the sender address, nonce, chain id and tx hash.
-// Note: 'msg' may contain fabricated information
-func VerifyEVMOutTxBody(msg *types.MsgAddToOutTxTracker, txBytes []byte, tssEth string) error {
-	var txx ethtypes.Transaction
-	err := txx.UnmarshalBinary(txBytes)
-	if err != nil {
-		return err
-	}
-	signer := ethtypes.NewLondonSigner(txx.ChainId())
-	sender, err := ethtypes.Sender(signer, &txx)
-	if err != nil {
-		return err
-	}
-	tssAddr := eth.HexToAddress(tssEth)
-	if tssAddr == (eth.Address{}) {
-		return fmt.Errorf("tss address not found")
-	}
-	if sender != tssAddr {
-		return fmt.Errorf("sender %s is not tss address", sender)
-	}
-	if txx.ChainId().Cmp(big.NewInt(msg.ChainId)) != 0 {
-		return fmt.Errorf("want evm chain id %d, got %d", txx.ChainId(), msg.ChainId)
-	}
-	if txx.Nonce() != msg.Nonce {
-		return fmt.Errorf("want nonce %d, got %d", txx.Nonce(), msg.Nonce)
-	}
-	if txx.Hash().Hex() != msg.TxHash {
-		return fmt.Errorf("want tx hash %s, got %s", txx.Hash().Hex(), msg.TxHash)
-	}
-	return nil
-}
-
-// VerifyBTCOutTxBody validates the SegWit sender address, nonce and chain id and tx hash
-// Note: 'msg' may contain fabricated information
-func VerifyBTCOutTxBody(msg *types.MsgAddToOutTxTracker, txBytes []byte, tssBtc string) error {
-	if !chains.IsBitcoinChain(msg.ChainId) {
-		return fmt.Errorf("not a Bitcoin chain ID %d", msg.ChainId)
-	}
-	tx, err := btcutil.NewTxFromBytes(txBytes)
-	if err != nil {
-		return err
-	}
-	for _, vin := range tx.MsgTx().TxIn {
-		if len(vin.Witness) != 2 { // outTx is SegWit transaction for now
-			return fmt.Errorf("not a SegWit transaction")
-		}
-		pubKey, err := btcec.ParsePubKey(vin.Witness[1], btcec.S256())
-		if err != nil {
-			return fmt.Errorf("failed to parse public key")
-		}
-		bitcoinNetParams, err := chains.BitcoinNetParamsFromChainID(msg.ChainId)
-		if err != nil {
-			return fmt.Errorf("failed to get Bitcoin net params, error %s", err.Error())
-		}
-		addrP2WPKH, err := btcutil.NewAddressWitnessPubKeyHash(
-			btcutil.Hash160(pubKey.SerializeCompressed()),
-			bitcoinNetParams,
-		)
+	if err != nil || tss == nil {
+		reason := "tss response is nil"
```

Comment:
I feel like this can be simplified into just returning an error if there is an error and then checking if TSS is nil.

Reply:
You mean removing the ErrTssNotFound wrapper? Why?